Add Panel Tabs react components

joe-s-avaya opened the pull request:
[Link to new Panel Tabs stories](https://deploy-preview-384--neo-react-library-storybook.netlify.app/?path=/docs/components-panel-tabs--docs) | [Link to Figma file](https://www.figma.com/file/1ewKUJ2JtiNwOSwFeCHCRk/branch/bpRUVH2YxlQ66U23cTluI2/Web-Components?type=design&node-id=11894-81225&mode=design&t=uMCcNV6ghuZ8TOz4-0)

**Before tagging the team for a review, I have done the following:**
- [x] reviewed my code changes
- [x] ensured that all tests pass
- [x] updated the link at the top of this PR (or remove it if not applicable)
- [x] added any important information to this PR (description, images, GIFs, ect.)
- [x] tagged Matt if any visual changes have occurred
- [x] done an accessibility check on my work

@enrique-prado:

- review stories for appropriate functionality
- review all files, but pay special attention to the `PanelTabs.tsx` file
- I will be adding the test file in the next PR (as this one is already huge)

@avaya-dux/dux-design:

- Review Agent Notes story for accuracy to Figma spec. This story does _not_ have any functionality (you can't add/edit/delete notes), but it should look correct.
- Review "Simple Panel Tabs" story for appropriate looks and interactions.


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

## Summary by CodeRabbit

- **New Features**
  - Introduced a new `PanelTabs` component for creating tabbed panels in applications, enhancing user interface organization and interaction.
  - Added example stories for `PanelTabs`, showcasing various usage scenarios like agent notes and simple tabs.

- **Style**
  - Implemented new styles for the `PanelTabs` component, ensuring visual consistency and appealing design across different color schemes and layouts.

- **Documentation**
  - Updated component stories to demonstrate the functionality and appearance of the new `PanelTabs` component.

- **Refactor**
  - Improved code readability in the `Pagination.stories.tsx` by adopting object destructuring for component parameters.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

matthewAvaya commented:
- When I am on the Notes tab and I select the collapse icon, I expect the Notes tab to be in its active state.
- Let's make all the expanded Panel Tabs to have the same fixed width as the agent view is much skinnier compared to the notes view. 
- The **Me** notes component (with light blue background) is missing the more menu (ellipses) icon as it should always be visible.
<img width="363" alt="Screen Shot 2024-05-08 at 3 24 25 PM" src="https://github.com/avaya-dux/neo-react-library/assets/90473510/974546eb-4587-4c89-b885-1c840f72cbfa">


- After selecting the expand/collapse button, it should no longer retain the hover state unless my cursor is still on it.

joe-s-avaya commented:
> * When I am on the Notes tab and I select the collapse icon, I expect the Notes tab to be in its active state.

@matthewAvaya talk to Justin about this one. We were given a specific request (I forget who from) that said the opposite. Thus, I will be leaving as is. 

> * Let's make all the expanded Panel Tabs to have the same fixed width as the agent view is much skinnier compared to the notes view.

That is on purpose. The Panel width is dynamic to its content. I've updated the "Agent Notes Example" story so that it's panels are all the same, but I have left the "Simple Panel Tabs" story as is so that it is obvious how to achieve both a static and dynamic width. 

> * The **Me** notes component (with light blue background) is missing the more menu (ellipses) icon as it should always be visible.

This is a static example, which is why no menu is available. I have implemented the dynamic examples in the Notes stories and do not want to bloat these stories with that functionality, so I am leaving them as is. 

> * After selecting the expand/collapse button, it should no longer retain the hover state unless my cursor is still on it.

That is the default Button functionality. When you click on a button, it receives "focus" and retains that focus until you move the focus to a different element (either via clicking or tabbing). I can hack the styling so that in these particular instances, there is no styling when the icon buttons have focus, but I do not recommend that action as it breaks from our default button functionality.

enrique-prado commented on src/components/PanelTabs/PanelTabs.stories.tsx:
```diff
@@ -0,0 +1,264 @@
+  render: () => {
+    const [index, setIndex] = useState(0);
+
+    return (
```
The outer panel size for this story behaves different than in the previous story. That in itself it's not a big deal but it raises the question as to how panels are being sized.  In this story, the contents (text) of the panel seems to drive the size of the panel. Notice how it changes depending on which is the active panel.  My concern is that if it behaves like this when this component is placed inside the Workspaces region then it will be problematic.  However, if it behaves like the previous story (it doesn't resize to the inner content) then that's what we want.
All this to say is that it should "just work" when it is placed inside a widget region.

joe-s-avaya replied:
updated

enrique-prado commented on src/components/PanelTabs/PanelTabs.css:
```diff
@@ -0,0 +1,91 @@
+:root .neo-paneltabs,
```
The rtl fix you put in place in the css library for the expand/collapse button is not here.  Did you plan to import it here at a later time?

joe-s-avaya replied:
Correct. After I merge the [CSS Panel Tabs update PR](https://github.com/avaya-dux/neo-css-library/pull/467) I'll be deleting this file and using the neo-css styling.

enrique-prado reviewed (commented):
Just a couple of questions, otherwise it's looking great!

joe-s-avaya commented on src/components/Pagination/Pagination.stories.tsx:
```diff
@@ -11,7 +11,13 @@ export default {
 } as Meta<PaginationProps>;
 
-const BookOfPages = ({ sectionWidth, direction="ltr" }: { sectionWidth?: number, direction?:string }) => {
+const BookOfPages = ({
```
You didn't format the file. This is a formatting change.

matthewAvaya commented:
- Can we add tooltip on the two icons, Notes and Agent, with the their supporting labels or is that handled by the product developers?
- On the Simple Panel Tabs page in Storybook, resize the width of the expanded pane to have the same widths as the other examples.

joe-s-avaya commented:
@matthewAvaya 

> * Can we add tooltip on the two icons, Notes and Agent, with the their supporting labels or is that handled by the product developers?

Up to the product devs. 

> * On the Simple Panel Tabs page in Storybook, resize the width of the expanded pane to have the same widths as the other examples.

Updated.

matthewAvaya commented:
- It looks like my previous comment didn't save. When the Panel Tabs is in collapsed view, allow the user to select the tab icons, such as for Notes and Agent, to expand the entire panel. That way the user doesn't have to move down to the bottom to expand the panel. Thanks!

joe-s-avaya commented:
> * It looks like my previous comment didn't save. When the Panel Tabs is in collapsed view, allow the user to select the tab icons, such as for Notes and Agent, to expand the entire panel. That way the user doesn't have to move down to the bottom to expand the panel. Thanks!

@matthewAvaya updated. @Justin-Avaya, please read Matt's comment above and comment if you disagree with this approach.

Justin-Avaya commented:
> > * It looks like my previous comment didn't save. When the Panel Tabs is in collapsed view, allow the user to select the tab icons, such as for Notes and Agent, to expand the entire panel. That way the user doesn't have to move down to the bottom to expand the panel. Thanks!
> 
> @matthewAvaya updated. @Justin-Avaya, please read Matt's comment above and comment if you disagree with this approach.

@joe-s-avaya I agree with @matthewAvaya comments and I would like to add that the UX team mentioned this morning, the user should not only have the ability to open the panel by clicking on a panel tab, but also clicking on an active panel tab should close the panel.

joe-s-avaya commented:
> the user should not only have the ability to open the panel by clicking on a panel tab, but also clicking on an active panel tab should close the panel.

Updated
